src/cascadia/WindowsTerminal/AppHost.cpp

```diff
@@ -78,40 +78,189 @@ AppHost::~AppHost()
+{
+    // get the command line
+    const wchar_t* const cmdLnPtr{ GetCommandLineW() };
+    if (!cmdLnPtr)
+    {
+        return false;
+    }
+
+    std::wstring cmdLn{ cmdLnPtr };
+    auto cmdLnEnd{ cmdLn.end() }; // end of still valid content in the command line
+    auto iter{ cmdLn.begin() }; // iterator pointing to the current position in the command line
+    auto argBegin{ iter }; // iterator pointing to the begin of an argument
+    auto quoted{ false }; // indicates whether a substring is quoted
+    auto within{ false }; // indicates whether the current character is inside of an argument
+
+    args.reserve(gsl::narrow_cast<size_t>(64U)); // pre-allocate some space for the string handles
+    while (iter < cmdLnEnd)
+    {
+        auto increment{ true }; // used to avoid iterator incrementation if a quotation mark has been removed
+        switch (*iter)
+        {
+        case L' ': // space and tab are the usual separators for arguments
+        case L'\t':
+            if (!quoted && within)
+            {
+                within = false;
+                args.emplace_back(argBegin, iter);
+            }
+            break;
+
+        case L'"': // quotation marks need special handling
+            std::move(iter + 1, cmdLnEnd, iter); // move the rest of the command line to the left to overwrite the quote
+            --cmdLnEnd; // the string isn't shorter but its valid content is, we can't just resize without potentially invalidate iterators
+            quoted = !quoted;
+            increment = false; // indicate that iter shall not be incremented because it already points to a new character
+        default: // any other character (including quotation mark)
+            if (!within)
+            {
+                within = true;
+                argBegin = iter;
+            }
+            break;
+        }
+
+        if (increment)
+        {
+            ++iter;
+        }
+    }
+
+    if (within)
+    {
+        args.emplace_back(argBegin, iter);
+    }
+
+    return true;
+}
+
+// Method Description:
+// - Take a file path and return the file name without file extension.
+// Arguments:
+// - path: A wstring_view containing the file path.
+// Return Value:
+// - A wstring_view containing the base name of the file.
+std::wstring_view AppHost::_GetBaseName(std::wstring_view path) const noexcept
+{
+    auto lastPoint = path.find_last_of(L'.');
+    const auto lastBackslash = path.find_last_of(L'\\');
+    auto lastSlash = path.find_last_of(L'/');
+    lastSlash = (lastBackslash != std::wstring::npos && lastSlash != std::wstring::npos) ? std::max(lastBackslash, lastSlash) : std::min(lastBackslash, lastSlash);
+    if ((lastSlash != std::wstring::npos && lastPoint < lastSlash) || lastPoint == std::wstring::npos)
+    {
+        lastPoint = path.size();
+    }
+
+    lastSlash = lastSlash == std::wstring::npos ? static_cast<size_t>(0U) : lastSlash + 1U;
+    return { &path.at(lastSlash), lastPoint - lastSlash };

```

Comment:
I'm fine with using `std::filesystem::path` here; it has convenience functions that make this a breeze

Reply:
Oh, I didn't discover the `stem` method yet.